src/app/backend/resource/pod/detail.go

```diff
@@ -202,47 +205,50 @@ func isNotFoundError(err error) bool {
 	return statusErr.ErrStatus.Code == 404
 }
 
-func toPodDetail(pod *v1.Pod, metrics []metric.Metric, configMaps *v1.ConfigMapList,
-	secrets *v1.SecretList, controller owner.ResourceOwner,
-	events *common.EventList) PodDetail {
-
+func extractContainerInfo(containerList []v1.Container, pod *v1.Pod, configMaps *v1.ConfigMapList, secrets *v1.SecretList) []Container {
 	containers := make([]Container, 0)
-	for _, container := range pod.Spec.Containers {
+	for _, c := range containerList {
```

Comment:
nit: please do not use single character variable names (exception if you want to indicate to the reader that the variable is unimportant and can be ignored, e.g. iterator i)